### PR TITLE
Components: refactor `ContextSystemProvider` and the `useUpdateEffect` util to ignore `exhaustive-deps` 

### PR DESCRIPTION
**Pull request by chad1008**
## What?
Updates the two `useUpdateEffect` utility hooks within `ContextSystemProvider` and `utils/hooks` to ignore the `exhaustive-deps` eslint rule

## Why?
The hook passes a non-array literal dependency list out of necessity (it's accepting the dependencies from the consumer and doesn't know what they are in advance). Further, the `effect` value, which is also passed in from the consumer is an additional missing dependency.

Adding `effect` to the deps array would require either a spread operator (which would trigger a different `exhaustive-deps` warning) or some more involved logic to handle the addition... but none of this would solve the original warning described above.

**Thoughts/questions for the future:**
1. These appear to be two identical hooks being declared in separate locations. That may be intentional, for reasons I'm not seeing, but if not do we want to refactor `ContextSystemProvider` to import the `useUpdateEffect` hook from utils? cc @sarayourfriend in case you have any insight 🙂 
2. Once we activate the rule for the Components package, would it make sense to [configure it to validate the dependencies of this custom hook](https://github.com/facebook/react/tree/main/packages/eslint-plugin-react-hooks#advanced-configuration)?

## How?
Added inline ignore comments to all `exhaustive-deps` warning in both hooks

### CODE REVIEW

**Comment by sarayourfriend** [comment]
> These appear to be two identical hooks being declared in separate locations. That may be intentional, for reasons I'm not seeing, but if not do we want to refactor ContextSystemProvider to import the useUpdateEffect hook from utils? cc @sarayourfriend in case you have any insight slightly_smiling_face

I rather suspect this was a mistake done in haste, rather than intentional. Alternatively, it may have been some workaround for the somewhat limited scope of the Gutenberg TypeScript configuration in early 2021. If the hooks are the same and there is no issue with TypeScript then I see no reason not to consolidate them and delete some code :smiling_imp:

File: packages/components/CHANGELOG.md [hunk 1]
```diff
@@ -49,6 +49,7 @@
 -   `Sandbox`: updated to satisfy `react/exhaustive-deps` eslint rule ([#44378](https://github.com/WordPress/gutenberg/pull/44378))
 -   `FontSizePicker`: Convert to TypeScript ([#44449](https://github.com/WordPress/gutenberg/pull/44449)).
 -   `FontSizePicker`: Replace SCSS with Emotion + components ([#44483](https://github.com/WordPress/gutenberg/pull/44483)).
+-   `Context`: updated to ignore `react/exhaustive-deps` eslint rule ([#45044](https://github.com/WordPress/gutenberg/pull/45044))
```
Comment (mirka) [inline packages/components/CHANGELOG.md]: Careful, this is not merging into the Unreleased section 😯

**Review by mirka — APPROVED** [review]
Yeah, seems like the disable is [inevitable](https://usehooks-ts.com/react-hook/use-update-effect).

> 2. Once we activate the rule for the Components package, would it make sense to [configure it to validate the dependencies of this custom hook](https://github.com/facebook/react/tree/main/packages/eslint-plugin-react-hooks?rgh-link-date=2022-10-17T19%3A03%3A49Z#advanced-configuration)?

I think so!

File: packages/components/CHANGELOG.md (hunk 1 above)
Comment (chad1008) [reply packages/components/CHANGELOG.md]: Yikes! Good catch, thank you! [Fixed](https://github.com/WordPress/gutenberg/pull/45044/commits/4a3aa8c377f98a22c9f939687026ac5ba3205e8b).

**Comment by chad1008** [comment]
Thanks @sarayourfriend  and @mirka!

I've gone ahead and removed the local `useUpdateEffect` in favor of the `utils` version. Still testing well for me (as it should, they were identical).

I'll update https://github.com/WordPress/gutenberg/pull/41166 shortly to include `useUpdateEffect` in the `exhaustive-deps` checks when we activate the rule for the full package.

@mirka, your approval was before the update to the `useUpdateEffect` hook, still good to go? 🙂

File: packages/components/src/ui/context/context-system-provider.js [hunk 2]
```diff
@@ -10,34 +10,20 @@ import {
 	createContext,
 	useContext,
 	useRef,
-	useEffect,
 	useMemo,
 	memo,
 } from '@wordpress/element';
 import warn from '@wordpress/warning';
+/**
```
Comment (ciampo) [inline packages/components/src/ui/context/context-system-provider.js:18]:
```suggestion

/**
```

**Comment by mirka** [comment]
Yep, still good! 🚢

File: packages/components/CHANGELOG.md [hunk 3]
```diff
@@ -37,7 +38,7 @@
 -   `withFocusReturn`: Refactor tests to `@testing-library/react` ([#45012](https://github.com/WordPress/gutenberg/pull/45012)).
 -   `ToolsPanel`: updated to satisfy `react/exhaustive-deps` eslint rule ([#45028](https://github.com/WordPress/gutenberg/pull/45028))
 -   `Tooltip`: updated to ignore `react/exhaustive-deps` eslint rule ([#45043](https://github.com/WordPress/gutenberg/pull/45043))
--   `SlotFill`: updated to satisfy `react/exhaustive-deps` eslint rule ([#44403](https://github.com/WordPress/gutenberg/pull/44403))
+-   `Context`: updated to ignore `react/exhaustive-deps` eslint rule ([#45044](https://github.com/WordPress/gutenberg/pull/45044))
```
Comment (ciampo) [inline packages/components/CHANGELOG.md]: This entry should also be moved to the `Unreleased` section

Comment (chad1008) [reply packages/components/CHANGELOG.md]: lol, I think I was noticing this and fixing it literally as you were leaving this comment.

Comment (ciampo) [reply packages/components/CHANGELOG.md]: image

**Pull request merged**